### PR TITLE
Fix issue #1145 - error lowering onnx.Cast

### PR DESCRIPTION
**Pull request by etiotto**
This PR fixes issue #1145. The model `gpt2-lm-head-10` for the model zoo will now compile correctly with this fix.

```
% time onnx-mlir -O3 ~/Downloads/gpt2-lm-head-10/GPT-2-LM-HEAD/model.onnx
onnx-mlir -O3 ~/Downloads/gpt2-lm-head-10/GPT-2-LM-HEAD/model.onnx  276.87s user 19.60s system 96% cpu 5:05.74 total
% echo $?
0
```

Signed-off-by: Ettore Tiotto <etiotto@ca.ibm.com>

### CODE REVIEW

**Review by tungld — APPROVED** [review]
LGTM!

File: src/Dialect/ONNX/MLIRDialectBuilder.cpp [hunk 1]
```diff
@@ -164,49 +167,76 @@ Value MathBuilder::select(Value cmp, Value lhs, Value rhs) const {
...
 
 Value MathBuilder::constantIndex(int64_t val) const {
   Attribute constantAttr = b.getIntegerAttr(b.getIndexType(), val);
   return b.create<arith::ConstantOp>(loc, constantAttr);
 }
 
-// For some reason, operations on unsigned int are often unhappy because
-// operations are mainly used on signless integers. So this cast remove the sign
-// of unsigned int for successful processing, to the best of my understanding.
+Value MathBuilder::createArithICmp(
```
Comment (AlexandreEichenberger) [inline src/Dialect/ONNX/MLIRDialectBuilder.cpp:209]: Could we overload `createArithCmp` (dropping the `I`) with the `arith::CmpIPredicate` for float, and a separate overloaded call for `arith::CmpFPredicate`

File: src/Dialect/ONNX/MLIRDialectBuilder.cpp [hunk 2]
```diff
@@ -164,49 +167,76 @@ Value MathBuilder::select(Value cmp, Value lhs, Value rhs) const {
...
+    Value lhs, Value rhs, arith::CmpIPredicate pred) const {
+  Type type = lhs.getType();
+  assert(type == rhs.getType() && "operands should have the same type");
+  assert((type.isa<IntegerType>() || type.isa<IndexType>()) &&
+         "Expecting IntegerType or IndexType");
+
+  if (type.isa<IndexType>())
```
Comment (AlexandreEichenberger) [inline src/Dialect/ONNX/MLIRDialectBuilder.cpp:216]: no need for a separate return value here. Please just convert the assert to check for signless or IndexType.

**Review by AlexandreEichenberger — CHANGES_REQUESTED** [review]
Requested minor changes

File: src/Dialect/ONNX/MLIRDialectBuilder.cpp (hunk 1 above)
Comment (etiotto) [reply src/Dialect/ONNX/MLIRDialectBuilder.cpp]: done

File: src/Dialect/ONNX/MLIRDialectBuilder.cpp (hunk 2 above)
Comment (etiotto) [reply src/Dialect/ONNX/MLIRDialectBuilder.cpp]: done

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, thanks

**Pull request merged**